File: Coding-test/Programmers/2022 KAKAO TECH INTERNSHIP/성격 유형 검사하기.js
function solution(survey, choices) {
  const score = {
    R: 0,
    T: 0,
    C: 0,
    F: 0,
    J: 0,
    M: 0,
    A: 0,
    N: 0,
  };

  choices.forEach((choice, idx) => {
    score[survey[idx][0]] += Math.max(4 - choice, 0);
    score[survey[idx][1]] += Math.max(choice - 4, 0);
  });

  let answer = "";
  answer += score["R"] >= score["T"] ? "R" : "T";
  answer += score["C"] >= score["F"] ? "C" : "F";
  answer += score["J"] >= score["M"] ? "J" : "M";
  answer += score["A"] >= score["N"] ? "A" : "N";

  return answer;
}

File: Coding-test/Programmers/2022 KAKAO TECH INTERNSHIP/성격 유형 검사하기.cpp
#include <cmath>
#include <map>
#include <string>
#include <vector>

using namespace std;

map<char, int> m;

string solution(vector<string> survey, vector<int> choices) {
  string answer = "";

  for (int i = 0; i < survey.size(); i++) {
    m[survey[i][0]] += max(4 - choices[i], 0);
    m[survey[i][1]] += max(choices[i] - 4, 0);
  }

  answer += m['R'] >= m['T'] ? "R" : "T";
  answer += m['C'] >= m['F'] ? "C" : "F";
  answer += m['J'] >= m['M'] ? "J" : "M";
  answer += m['A'] >= m['N'] ? "A" : "N";

  return answer;
}